<div style="width: 100%; text-align: center; color: white; background-color: #222; font-size: 3em; font-weight: 700; padding: .5em .5em .6em .5em; line-height: 1.4em;">Programação em Python para Iniciantes</div>

<br>

<!-- # Programação em Python para Iniciantes -->

Antes de mais nada, este é um _nootebook_: um arquivo estruturado que contém texto e código intercalados. A maneira ideal de acessar este arquivo, seja para leitura ou edição, é através do software [Jupyter](http://jupyter.org/). 

Em um _notebook_, as unidades básicas de texto ou de código são chamadas de células. É fácil distinguir as células de código: elas vem precedidas por 

    In [ ]:
    
como a célula abaixo.

In [1]:
2 + 2

4

Se você estiver acessando este arquivo pelo Jupyter, então muito provavelmente você poderá executar as células de código. Para executar uma célula de código, basta selecionar a célula e digitar <kbd>SHIFT+ENTER</kbd>. Qualquer saída gerada pelo código executado será exibida logo em seguida.

Neste _notebook_, vamos aprender alguns elementos de programação em Python e vamos nos familiarizar com o ambiente do Jupyter.

<hr style="border-color: #ddd;">

# Python como uma calculadora

Em qualquer trabalho moderno, é comum precisarmos de uma calculadora para fazer contas. O console do Python é uma excelente ferramenta para se fazer cálculos.

## As quatro operações

Clique em cada célula abaixo, tecle <kbd>SHIFT-ENTER</kbd> e veja o que acontece.

In [2]:
10 + 5 - 3

12

In [3]:
4 * 3

12

In [4]:
2800.20 / 133.21598477640342

21.02

In [5]:
12 / 5

2.4

Ou seja, usamos os símbolos &nbsp;`+`&nbsp;, &nbsp;`-`&nbsp;, &nbsp;`*`&nbsp; e &nbsp;`/`&nbsp; para soma, subtração, multiplicação e divisão, respectivamente. Isso é como você esperava, não é?

Então, para executar uma célula, basta teclar <kbd>SHIFT-ENTER</kbd>. Se você teclar <kbd>ENTER</kbd> sozinho, você criará mais linhas dentro da mesma célula.

<span style="color: #FF0000;"><b>Atenção:</b></span> o caractére que separa as casas decimais em Python é o **ponto** e não a vírgula!

## Potenciação

A potenciação é denotada por `**`. Ou seja, para calcular $2^4$, precisamos escrever `2**4`.

Execute as células abaixo apertando <kbd>SHIFT-ENTER</kbd>. Tente prever o resultado antes!

In [6]:
2**4

16

In [7]:
3**2

9

## Divisão inteira e resto

Python tem um tipo especial de divisão que é chamada de divisão inteira, em que um número é dividido por outro e depois o resultado é **arrendondado** para baixo. O operador de divisão inteira é `//`.

Mais formalmente, quando `a` e `b` são números inteiros, `a // b` é o quociente da divisão de `a` por `b`.

Veja como a divisão inteira funciona:

In [8]:
3//2

1

In [9]:
11//3

3

Python também tem o operador &nbsp;`%`&nbsp; que calcula o resto da divisão de um número por outro.

Por exemplo, o resto da divisão de 17 por 5 é igual 2. Veja:

In [13]:
17%5

2

In [14]:
13%7

6

## Combinando operações

Você consegue prever o resultado das expressões abaixo?

Execute as células para verificar se você acertou!


In [15]:
8*3//2**2

6

In [17]:
(1+5)*3

18

Neste último caso, o Python primeiro calcula o `5*3` e depois soma 1, certo?

Isso acontece porque o Python executa as operações na seguinte ordem de precedência:


1.    &nbsp;&nbsp;&nbsp;`**`
2.    &nbsp;&nbsp;&nbsp;`*`&nbsp;&nbsp;&nbsp;&nbsp;`/`&nbsp;&nbsp;&nbsp;&nbsp;`//`&nbsp;&nbsp;&nbsp;&nbsp;`%`
3.    &nbsp;&nbsp;&nbsp;`+`&nbsp;&nbsp;&nbsp;&nbsp;`-`

Na dúvida, você pode colocar parênteses para deixar a ordem explícita.

Poderíamos, por exemplo, alterar a célula acima colocando `(1+5)*3` para indicar que o `+` deve ser feito antes do `*`.



<hr>

### Exercício <span class="exec"></span>: parcelas sem juros I

Alice pagou uma televisão de R$ 3250.00 em 12 vezes sem juros. Qual o valor de cada parcela?

In [18]:
3250/12

270.8333333333333

<hr>

### Exercício <span class="exec"></span>: aniversário

João faz 10 anos de idade hoje, neste exato momento! Na célula abaixo, calcule quantos segundos de vida ele acabou de comemorar. (Ignore os anos bissextos.)


In [21]:
60*60*24*365*10

315360000

<hr>

### Exercício <span class="exec"></span>: vazamento

Uma torneira pinga a uma taxa de 1 gota por segundo. Cada gota possui um volume de 0.05 mililitros. Calcule o volume de água desperdiçado depois de exatos 3 dias de vazamento. Faça a conta de modo que a resposta seja em litros.

In [22]:
((1*60*60*24*3)*0.05)*(10**(-3))

12.96

---

### Exercício <span class="exec"></span>: dobradura

Uma folha de papel sulfite tem 0.1mm de espessura. Dobrando a folha ao meio, a espessura passa a ser 0.2mm. Dobrando ao meio de novo, a espessura vai para 0.4mm. Mais uma vez e a espessura aumenta para 0.8mm, e assim vai. 

Sabemos que não é possível dobrar uma folha de papel sulfite ao meio mais do que 5 ou 6 vezes mas, supondo que fosse possível dobrá-la 27 vezes, qual seria a espessura da dobradura resultante?

Faça a conta de modo que a resposta seja em kilômetros.

In [32]:
((2**27)*0.1)*(10**(-6))

13.4217728



## Variáveis

Você pode guardar o resultado de um cálculo em uma variável para tornar a usá-lo.

Vamos fazer o cálculo de conversão de temperatura de Celsius para Fahrenheit. Podemos começar colocando a temperatura em Celsius, digamos 55 graus, numa variável chamada `C`.

Não se esqueça de teclar <kbd>SHIFT+ENTER</kbd> em cada uma das células abaixo.

In [41]:
C = 80

Veja o valor de C teclando <kbd>SHIFT+ENTER</kbd> na célula abaixo.

In [42]:
C

80

Para converter para Fahrenheit, temos que multiplicar `C` por `9/5` e depois somar `32`:

In [43]:
F = C * 9 / 5 + 32

Veja o valor de F teclando <kbd>SHIFT+ENTER</kbd> na célula abaixo:

In [45]:
F

176.0


E se você quisesse converter &nbsp;`65`&nbsp; em vez de &nbsp;`55`?

Basta voltar na linha em que você definiu &nbsp;`C = 55`&nbsp; e mudar para &nbsp;`C = 65`&nbsp; e reexecutar a célula.

Em seguida, reexecute a linha que definiu &nbsp;`F = C \* 9 / 5 + 32`&nbsp; para atualizar o valor em Fahrenheit.

Depois, reexecute a linha seguinte para imprimir o novo valor de &nbsp;`F`.

O novo valor de &nbsp;`F`&nbsp; deve ser &nbsp;`149`.


<hr>

### Exercício <span class="exec"></span>: conversão de temperatura


Modifique as células acima (e reexecute tudo o que for necessário) para converter 80 Celsius para Fahrenheit!

<hr>

#### Maiúscula ou minúscula?

<span style="color: #FF0000;"><b>Atenção:</b></span> letras maiúsculas ou minúsculas fazem diferença!

Uma variável chamada `C` e outra chamada `c` são coisas diferentes! Veja o que acontece:


In [46]:
C = 0
c = 1

In [47]:
C

0

In [48]:
c

1

Como uma boa prática de programação, evite usar acentos ou outros caractéres especiais em nomes de variáveis.

<span style="color: #FF0000;"><b>Atenção:</b></span> nome de variável não pode começar com dígito numérico, mas pode conter dígitos depois do primeiro caractére.


# Mais sobre o Jupyter


### Múltiplas linhas de código

Células de código podem ter múltiplas linhas. Veja como ficaria o exemplo da conversão de unidades de temperatura numa única célula:

In [49]:
C = 55
F = C * 9 / 5 + 32
F

131.0

### Salvando seu trabalho

Para salvar o seu trabalho, cliquem em **_File &raquo; Save and Checkpoint_** ou no ícone do disquete.

**Atenção:** se você estiver usando o site [tmpnb.org](http://tmpnb.org), não é possível salvar arquivos! Nesse caso, clique em **_File &raquo; Download as &raquo; IPython Notebook (.ipynb)_** para baixar uma cópia do _notebook_ para o seu computador, e baixe cópias frequentemente para não perder o seu trabalho!


### Células de texto

Você pode modificar os textos explicativos e fazer _seus próprios comentários_ em [Markdown](https://en.wikipedia.org/wiki/Markdown). Para editar uma célula de texto (como esta) é só dar um duplo-clique na célula. Quando terminar de editar, é só apertar <kbd>SHIFT-ENTER</kbd> para que o texto fique bem formatado novamente. <i>Tags</i> <span style="color: #c94c4c;"><b>HTML</b></span> e fórmulas em _Latex_ também são permitidas:

$$ \int_0^t 3x^2\, dx $$


### Mais sobre células

Explore os menus _**Edit**_, _**Insert**_ e depois _**Cell**_, para ver quais operações com células podem ser efetuadas. Você também pode usar atalhod pelo teclado para acessar essas funções:

* <kbd>ESC B</kbd> insere uma nova célula de código **abaixo** da célula selecionada;
* <kbd>ESC A</kbd> insere uma nova célula de código **acima** da célula selecionada;
* <kbd>ESC M</kbd> transforma uma célula em célula de **texto**;
* <kbd>ESC Y</kbd> transforma uma célula em célula de **código**;
* <kbd>CTRL+SHIFT+P</kbd> mostra mais atalhos do teclado.




# Entrada e saída


Todo programa de computador precisa:

* receber dados de alguma fonte (teclado, mouse, arquivo, internet, etc.), 
* processar esses dados e, 
* enviar o resultado do processamento para algum lugar (tela, arquivo, impressora, web, etc.).

Então, quando tratamos do fluxo de dados por um programa de computador, é comum classsificar os dados como

* dados de entrada: são aqueles que o programa **recebe**, e
* dados de saída: são aqueles que programa **mostra** ao usuário.



## Mostrando dados (saída)


No exemplo de conversão de temperatura, nós usamos o seguinte código.

In [50]:
C = 55
F = C * 9 / 5 + 32
F

131.0

Suponha agora que também quiséssemos calcular a temperatura em Kelvin:

In [51]:
C = 55
F = C * 9 / 5 + 32
F
K = C + 273.15
K

328.15

Hummm... só o K apareceu e nada do F!

Para fazer com que um dado seja mostrado na saída, podemos usar a função `print()`:

In [52]:
C = 55
F = C * 9 / 5 + 32
print(F)
K = C + 273.15
print(K)

131.0
328.15


Agora sim!

Poderíamos também deixar a saída melhor formatada, você não acha?

In [53]:
C = 55
F = C * 9 / 5 + 32
print(C, "Celsius em Fahrenheit é", F)
K = C + 273.15
print(C, 'Celsius em Kelvin é', K)

55 Celsius em Fahrenheit é 131.0
55 Celsius em Kelvin é 328.15


Em linhas gerais, a função `print()` recebe uma lista de argumentos (dados numéricos ou de texto) e mostra esses argumentos na saída, separando-os com um espaço em branco. Por fim, o `print()` manda "pular uma linha" na saída.

## Perguntando dados (entrada)


Agora que você já sabe mostrar dados na saída, é importante que você aprenda a fazer perguntas ao usuário e saiba receber a reposta que ele digitar.

Para pedir que o usuário entre com um dado de texto, use a função `input()`. 

Veja os exemplos:


In [54]:
nome = input("Qual é seu nome? ")
print("Olá,", nome)

Qual é seu nome? Guilherme
Olá, Guilherme


Mas olha agora que estranho, execute o código abaixo e responda 1 quando for perguntado a respeito da idade:

In [55]:
idade = input("Qual é a sua idade? ")
dobro = idade + idade
print("O dobro da sua idade é", dobro)

Qual é a sua idade? 1
O dobro da sua idade é 11


**11?? Mas por quê??**

O que acontece é que a função `input()` do Python sempre devolve o que o usuário digitar em formato **texto**. 

Em Python, textos são delimitados por aspas simples ou duplas. Veja:

In [56]:
"Maria"

'Maria'

In [57]:
"João"

'João'

E olha o que acontece com a soma de dois textos:

In [58]:
"Maria" + "João"

'MariaJoão'

Ou seja, somar dois textos tem o efeito de grudá-los.

Para ser um pouco mais formal, nós dizemos que o operador `+` **concatena** os textos.

Então, no exemplo anterior, como o Python entendeu que a entrada era um texto, o que ele fez foi equivalente ao seguinte:

In [60]:
idade = "123"
dobro = idade + idade 
print(dobro)

123123


---

### Exercício <span class="exec"></span>: idade avançada

Mude apenas o valor inicial da variável `idade` no programa acima para imprimir 123123 em vez de 11.

---

## Conversão texto $\rightarrow$ número

Mas o que fazer se quisermos perguntar números em vez de texto?

É importante que você saiba que, intenamente, existem três tipos de dados numéricos no Python 3:

* números inteiros ( *int* ), como 1, 0, -1, 2, 4
* números de ponto flutuante ( *float* ), como 2.13, 1.1, 1.00001, -12.3
* números complexos.


Para pedir um número ao usuário, basta converter o texto que ele digitou para um número do tipo desejado. Isso pode ser feito através das funções `int()` ou `float()`. Veja um exemplo abaixo.

In [61]:
idade_texto = input("Qual é a sua idade? ")
idade = int(idade_texto)
dobro = idade + idade
print("O dobro da sua idade é", dobro)

Qual é a sua idade? 19
O dobro da sua idade é 38


A linha 

    idade = int(idade_texto)
        
converte o valor guardado na variável `idade_texto` para o tipo _int_ e guarda o resultado em `idade`.

Escolhemos converter para _int_ pois consideramos idade como sendo um número inteiro.

É mais comum &ndash; e mais desejável &ndash; escrever o programa assim:

In [62]:
idade = int(input("Qual é a sua idade? "))
dobro = idade + idade
print("O dobro da sua idade é", dobro)

Qual é a sua idade? 19
O dobro da sua idade é 38


Nessa versão do programa, a linha 

    idade = int(input("Qual é a sua idade? "))
    
faz o mesmo que as duas linhas 

    idade_texto = input("Qual é a sua idade? ")
    idade = int(idade_texto)

e tem a vantagem de deixar o código mais sucinto e mais legível.

<hr>

Se você quiser permitir que a idade seja um número quebrado, você pode converter o texto lido para _float_:

In [63]:
idade = float(input("Qual é a sua idade? "))
dobro = idade + idade
print("O dobro da sua idade é", dobro)

Qual é a sua idade? 19
O dobro da sua idade é 38.0


## Conversão número $\rightarrow$ texto

Para converter um número (ou uma expressão numérica) para texto, você pode usar a função `str()`.

Veja o exemplo:

In [65]:
ano = int(input('Digite um ano qualquer do passado (4 dígitos): '))
idade = int(input('Em ' + str(ano) + ', quantos anos você completou? '))
print('Em 2016, você completou ' + str(idade + 2016 - ano) + ' anos.')

Digite um ano qualquer do passado (4 dígitos): 2020
Em 2020, quantos anos você completou? 19
Em 2016, você completou 15 anos.


### Limitando as casas decimais

Para exibir somente um número limitado de casas decimais de um número _float_, podemos usar a função `round()`. Veja o exemplo abaixo.

In [66]:
round(234.234812312854, 3)

234.235

Veja outro exemplo.

In [67]:
valor = 1573.90
num_parcelas = 13
print('O valor de cada parcela será de R$ ' + str(round(valor/num_parcelas, 2)))

O valor de cada parcela será de R$ 121.07


### Formatação de texto

Para mais informações sobre formatação de número ou texto, consulte a página [https://pyformat.info/](https://pyformat.info/).

# Mais exercícios

---

### Exercício <span class="exec"></span>: parcelas sem juros II

Escreva um programa que auxilia uma loja na hora de fazer uma venda. Seu programa deve:

* perguntar o preço do produto;
* perguntar o número de parcelas em que o cliente deseja pagar (sem juros);
* mostrar quanto será o valor de uma parcela.

Obs.: lembre-se de digitar os número com ponto (como 18.50 ao invés de 18,50).

In [15]:
productPrice = float(input("Informe o preço do produto: "))
installmentsNumber = int(input("Informe o número de parcelas que deseja pagar (sem juros): "))

print("==================================================================")
print("Cada parcela terá o valor de R$" + str(round(productPrice / installmentsNumber, 2)))

Informe o preço do produto: 1220
Informe o número de parcelas que deseja pagar (sem juros): 12
Cada parcela terá o valor de R$101.67


---

### Exercício <span class="exec"></span>: autonomia

Faça um programa que pergunta ao usuário pela capacidade do tanque de gasolina do carro (em litros), depois pelo consumo de gasolina (em km/l). Depois, seu programa deve mostrar a autonomia do carro com o tanque cheio, ou seja, quantos kilômetros ele consegue viajar com o tanque cheio até parar.

In [14]:
tankCapacity = float(input("Informe a capacidade do tanque de gasolina (em litros): "))
gasolineConsumption = float(input("Informe o consumo de gasolina do veículo (em km/l): "))

print("==================================================================")
print("O carro consegue viajar " + str(round(tankCapacity * gasolineConsumption, 1)) + "km com o tanque cheio.")

Informe a capacidade do tanque de gasolina (em litros): 12
Informe o consumo de gasolina do veículo (em km/l): 2
O carro consegue viajar 24.0km com o tanque cheio.


---

### Exercício <span class="exec"></span>: área do triângulo

Faça um programa para calcular a área de um triângulo. Seu programa deve pedir para o usuário digitar o comprimento da base e depois a altura, e seu programa diz qual a área de um triângulo com tais parâmetros.

In [13]:
length = float(input("Informe o comprimento da base do triângulo (em cm): "))
height = float(input("Informe a altura do triângulo (em cm): "))

print("==================================================================")
print("O triângulo tem uma área de " + str(round(((length * height) / 2), 2)) + " cm.")

Informe o comprimento da base do triângulo (em cm): 12
Informe a altura do triângulo (em cm): 3
O triângulo tem uma área de 18.0 cm.


---

### Exercício <span class="exec"></span>: média ponderada

Faça um programa que 

* pergunta a um aluno quanto ele tirou na P1, depois quanto ele tirou na P2; e
* mostra na saída a média ponderada, considerando que o peso da P1 é 2.5 e o peso da P2 é 4.0 (portanto a soma dos pesos é 6.5).

In [12]:
p1 = float(input("Informe a nota da P1: "))
p1Weight = 2.5
p2 = float(input("Informe a nota da P2: "))
p2Weight = 4.0
average = (p1 * p1Weight + p2 * p2Weight) / (p1Weight + p2Weight)

print("==================================================================")
print("A média ponderada do aluno é", round(average, 2))

Informe a nota da P1: 5
Informe a nota da P2: 7
A média ponderada do aluno é 6.23


---

### Exercício <span class="exec"></span>: volume do cilíndro

Peça ao usuário para digitar dois números: o raio $r$ da base do cilindro e sua altura $h$, um por vez. Depois mostre na tela o volume do cilindro, que pode ser calculado com a fórmula

$$ V = \pi r^2 h. $$

Use o valor 3.1415926 para aproximar $\pi$.

In [11]:
radius = float(input("Informe o raio da base do cilindro (em cm): "))
height = float(input("Informe a altura do cilindro (em cm): "))
pi = 3.1415926

volume = pi * (radius**2) * height
print("==================================================================")
print("O volume do cilindro é", round(volume, 2), "cm cúbicos.")

Informe o raio da base do cilindro (em cm): 4
Informe a altura do cilindro (em cm): 4
O volume do cilindro é 201.06 cm cúbicos.


---

### Exercício <span class="exec"></span>: volume da esfera

O cálculo do volume de uma esfera de raio $R$ é feito pela fórmula

$$ V = \frac{4 \pi R^3}{3}. $$

Faça um programa que pergunta ao usuário por um número positivo $R$ e que calcula e depois mostra na tela o volume de uma esfera de raio $R$.

Use o valor 3.1415926 para aproximar $\pi$.

In [10]:
R = float(input("Informe o raio da esfera (em cm): "))
V = (4 * pi * R**3) / 3
print("==================================================================")
print("O volume de uma esfera de raio " + str(R) + " cm é " + str(round(V, 2)) + " cm cúbicos.")

Informe o raio da esfera (em cm): 50
O volume de uma esfera de raio 50.0 cm é 523598.77 cm cúbicos.


---

### Exercício <span class="exec"></span>: poupança

Um investidor depositou uma quantia na poupança, que rende $p$ porcento ao mês. Ajude o investidor a calcular quanto ele terá depois de alguns meses. 

Seu programa deve:

* perguntar pela quantia $d$ de dinheiro depositada na poupança;
* perguntar pela taxa $p$ de rendimento da poupança;
* perguntar pelo numero $t$ de meses que o dinheiro ficará rendendo;
* mostrar o montante de dinheiro na poupança após os $t$ meses.

Lembre-se que a quantia $D$ após $t$ meses pode ser calculada por

$$ D = d\Big(1 + \frac{p}{100}\Big)^t. $$

In [7]:
d = float(input("Quanto foi depositado na poupança? "))
p = float(input("Qual a taxa de rendimento da poupança? "))
t = int(input("Quanto tempo em meses deixará o dinheiro rendendo? "))

D = d * (1 + p / 100) ** t

print("==================================================================")
print("Após os " + str(t) + " meses rendendo na poupança, o montante será de R$" + str(round(D, 2)) + ".")

Quanto foi depositado na poupança? 100
Qual a taxa de rendimento da poupança? 5.15
Quanto tempo em meses deixará o dinheiro rendendo? 12
Após os 12 meses rendendo na poupança, o montante será de R$182.69.


---

### Exercícios <span class="exec"></span>: horas, minutos e segundos

O cliente de uma empresa gostaria de um programa que convertesse uma quantidade $S$ de segundos em horas, minutos e segundos. Faça esse programa.

**Dica:** use os operadores de divisão inteira `//` e resto `%`.

In [5]:
S = int(input("Informe os segundos que serão convertidos: "))
hours = (S // 60) // 60
minutes = (S // 60) % 60
seconds = S % 60
print("==================================================================")
print(str(hours) + " hora(s), " + str(minutes) + " minuto(s) e " + str(seconds) + " segundo(s).")

Informe os segundos que serão convertidos: 3662
1 hora(s), 1 minuto(s) e 2 segundo(s).


---

### Exercício <span class="exec"></span>: troco

Você é comerciante e identificou um problema muito comum: uma pessoa compra um produto de sua loja mas, na hora de passar o cartão dela na maquininha, o caixa cobra $C \geq 0$ centavos a mais do que deveria e agora precisa dar o troco. Sua loja só trabalha com moedas ou cartão, então o cliente tem direito a receber _em moedas_ a quantidade $C$ de centavos que excedeu o preço da compra. Você quer dar ao cliente o número mínimo de moedas para trocar o valor $C$.

Por exemplo, se o caixa cobrou 573 centavos a mais no cartão, o cliente tem direito a receber

    5 moeda(s) de  1 real
    1 moeda(s) de 50 centavos
    0 moeda(s) de 25 centavos
    2 moeda(s) de 10 centavos
    0 moeda(s) de  5 centavos
    3 moeda(s) de  1 centavo

Faça um programa para ajudá-lo nesta tarefa:

* pergunte ao usuário pela quantidade `C` de centavos que deve ser trocada;
* mostre ao usuário quais moedas de cada denominação devem ser devolvidas ao cliente em um troco mínimo.

**Dica:** use os operadores de divisão inteira `//` e resto `%`.

In [4]:
C = int(input("Informe quantos centavos devem ser trocados: "))
print("==================================================================")
print(C // 100, "moeda(s) de 1 real")
print(C % 100 // 50, "moeda(s) de 50 centavos")
print(C % 100 % 50 // 25, "moeda(s) de 25 centavos")
print(C % 100 % 50 % 25 // 10, "moeda(s) de 10 centavos")
print(C % 100 % 50 % 25 % 10 // 5, "moeda(s) de 5 centavos")
print(C % 100 % 50 % 25 % 10 % 5, "moeda(s) de 1 centavo")

Informe quantos centavos devem ser trocados: 573
5 moeda(s) de 1 real
1 moeda(s) de 50 centavos
0 moeda(s) de 25 centavos
2 moeda(s) de 10 centavos
0 moeda(s) de 5 centavos
3 moeda(s) de 1 centavo


### Exercícios (URI Judge)

#### Exercício 01: Hello World!

In [3]:
print("Hello World!")

Hello World!


#### Exercício 02: Extremely Basic

In [2]:
A = int(input())
B = int(input())
X = A + B
print("X =", X)

10
9
X = 19


#### Exercício 03: Area of a Circle

In [6]:
R = float(input())
A = 3.14159 * (R ** 2)
print("A=" + format(A,".4f"))

150.00
A=70685.7750


#### Exercício 04: Simple Sum

In [7]:
A = int(input())
B = int(input())
SOMA = A + B
print("SOMA =", SOMA)

3
2
SOMA = 5


#### Exercício 05: Simple Product

In [8]:
number1 = int(input())
number2 = int(input())
PROD = number1 * number2
print("PROD =", PROD)

3
4
PROD = 12


#### Exercício 06: Average 1

In [5]:
A = float(input())
B = float(input())
MEDIA = (A * 3.5 + B * 7.5) / (3.5 + 7.5)
print("MEDIA = " + format(MEDIA, ".5f"))

5.0
7.1
MEDIA = 6.43182


#### Exercício 07: Difference

In [7]:
A = int(input())
B = int(input())
C = int(input())
D = int(input())
DIFERENCA = (A * B - C * D)
print("DIFERENCA =", DIFERENCA)

5
6
7
8
DIFERENCA = -26


#### Exercício 08: Simple Calculate

In [19]:
product1 = input()
product2 = input()
[code1, units1_str, price1_str] = product1.split(' ')
[code2, units2_str, price2_str] = product2.split(' ')

units1_num = int(units1_str)
units2_num = int(units2_str)
price1_num = float(price1_str)
price2_num = float(price2_str)

amount = (units1_num * price1_num) + (units2_num * price2_num)
print("VALOR A PAGAR: R$ " + format(amount, ".2f"))

1 1 15.10
2 1 15.10
VALOR A PAGAR: R$ 30.20


#### Exercício 09: Distance Between Two Points

In [22]:
p1 = input()
p2 = input()

[x1_str, y1_str] = p1.split(' ')
[x2_str, y2_str] = p2.split(' ')

x1 = float(x1_str)
y1 = float(y1_str)
x2 = float(x2_str)
y2 = float(y2_str)

distance = ((x2 - x1) ** 2 + (y2 - y1) ** 2) ** (1/2)

print(format(distance, ".4f"))

2.5 -0.4
-12.2 7.0
16.4575


#### Exercício 10: Fuel Spent

In [13]:
tankCapacity = 12
spentTime = int(input())
averageSpeed = int(input())

distance = averageSpeed * spentTime
liters = distance / tankCapacity

print(format(liters, ".3f"))

22
67
122.833


<span style="color: red;">Ignore as próximas células! Elas só estão presentes para personalizar a formatação deste _notebook_. </span>

In [ ]:
%%html
<style>

.rendered_html h1 {
    background-color: #555;
    color: white;
    padding: .5em;
    // border-bottom: 2px solid #000;
    // padding-bottom: .6em;
    margin-bottom: 1em;
}

.rendered_html h2 {
    border-bottom: 1px solid #333;
    padding-bottom: .6em;
}
                                      
.rendered_html h3 {
    color: #034f84;
}

.rendered_html code  {
    padding: 2px 4px;
    font-size: 90%;
    color: #c7254e;
    background-color: #f9f2f4;
    border-radius: 4px;
}

.rendered_html pre code {
    padding: 0px;
    font-size: 90%;
    color: #c7254e;
    background-color: rgba(0, 0, 0, 0);
}

kbd {
    border-radius: 3px;  
    padding: 2px, 3px;
}

body {
    counter-reset: h1counter excounter;
}
h1:before {
    content: counter(h1counter) ".\0000a0\0000a0";
    counter-increment: h1counter;
}
span.exec:before {
    content: counter(excounter);
    counter-increment: excounter;
}

</style>                                      